src/redis/rax.c

```diff
@@ -1221,29 +1221,44 @@ int raxRemove(rax *rax, unsigned char *s, size_t len, void **old) {
 
 /* This is the core of raxFree(): performs a depth-first scan of the
  * tree and releases all the nodes found. */
-void raxRecursiveFree(rax *rax, raxNode *n, void (*free_callback)(void*)) {
+void raxRecursiveFree(rax *rax, raxNode *n, void (*free_callback)(void*, void*), void* argument) {
     debugnode("free traversing",n);
     int numchildren = n->iscompr ? 1 : n->size;
     raxNode **cp = raxNodeLastChildPtr(n);
     while(numchildren--) {
         raxNode *child;
         memcpy(&child,cp,sizeof(child));
-        raxRecursiveFree(rax,child,free_callback);
+        raxRecursiveFree(rax,child,free_callback,argument);
         cp--;
     }
     debugnode("free depth-first",n);
     if (free_callback && n->iskey && !n->isnull)
-        free_callback(raxGetData(n));
+        free_callback(raxGetData(n),argument);
     rax_free(n);
     rax->numnodes--;
 }
 
+/* Free the entire radix tree, invoking a free_callback function for each key's data. 
+ * An additional argument is passed to the free_callback function.*/
+ void raxFreeWithCallbackAndArgument(rax *rax, void (*free_callback)(void*, void*), void* argument) {
```

Comment:
Should we call this:

```suggestion
 void raxRecursiveFreeWithCallbackAndArgument(rax *rax, void (*free_callback)(void*, void*), void* argument) {
```

?

Reply:
I don't think so, because Redis provides `raxFreeWithCallback` as a public API. And `raxRecursiveFree` is an internal method and not part of the public API. So for consistency, I think `raxFreeWithCallbackAndArgument` would be a better choice.

Reply:
ok :shrug: